rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -161,8 +164,9 @@ void SequentialWriter::close()
     metadata_io_->write_metadata(base_folder_, metadata_);
   }
 
-  storage_.reset();  // Necessary to ensure that the storage is destroyed before the factory
-  storage_factory_.reset();
+  // destroy current storage object and clear active track map for topic metadata
+  storage_.reset();
+  topics_names_to_info_.clear();
```

Comment:
how about `topic_names_to_message_definitions_` i think this should be cleared at the same time here?
IMO, we need to clear the all member variables that are not reset in the open method for the fresh start.